src/main/scala/com/actian/spark_vector/vector/Vector.scala

```diff
@@ -18,72 +18,81 @@ package com.actian.spark_vector.vector
 import scala.concurrent.Await
 import scala.concurrent.duration.Duration
 
-import org.apache.spark.scheduler.{ SparkListener, SparkListenerJobEnd }
-import org.apache.spark.Logging
+import org.apache.spark.{ Logging, SparkContext }
 import org.apache.spark.rdd.RDD
-import org.apache.spark.sql.types.StructType
+import org.apache.spark.scheduler.{ SparkListener, SparkListenerJobEnd }
 import org.apache.spark.sql.Row
-import org.apache.spark.SparkContext
-import org.apache.spark.ContextCleaner
+import org.apache.spark.sql.types.StructType
 
-import com.actian.spark_vector.util.{ RDDUtil, ResourceUtil }
-import com.actian.spark_vector.datastream.DataStreamClient
-import com.actian.spark_vector.datastream.writer.{ DataStreamWriter, InsertRDD }
+import com.actian.spark_vector.datastream.{ DataStreamClient, VectorEndpointConf }
 import com.actian.spark_vector.datastream.reader.{ DataStreamReader, ScanRDD }
+import com.actian.spark_vector.datastream.writer.{ DataStreamWriter, InsertRDD }
+import com.actian.spark_vector.util.{ RDDUtil, ResourceUtil }
 
 /** Utility object that defines methods for loading data into Vector */
 private[vector] object Vector extends Logging {
+  import VectorUtil._
   import RDDUtil._
   import ResourceUtil._
-  import VectorUtil._
 
-  /**
-   * Given an `rdd` with data types specified by `schema`, try to load it to the Vector table `targetTable`
-   * using the connection information stored in `vectorProps`.
+  private def prepareRDD(rdd: RDD[Seq[Any]],
+    rddSchema: StructType,
+    targetSchema: StructType,
+    fieldMap: Option[Map[String, String]] = None): RDD[Seq[Any]] = {
+    val resolvedFieldMap = fieldMap.getOrElse(Map.empty)
+    // Apply the given field map return a sequence of field name, column name tuples
+    val field2Columns = applyFieldMap(resolvedFieldMap, rddSchema, targetSchema)
+
+    // Validate the list of columns are OK to load
+    validateColumns(targetSchema, field2Columns.map(_.columnName))
+
+    // If a subset of input fields are needed to load, select only the fields needed
+    val (inputRDD, inputType) =
+      if (field2Columns.length < rddSchema.fields.length) {
+        selectFields(rdd, rddSchema, field2Columns.map(_.fieldName))
+      } else {
+        (rdd, rddSchema)
+      }
+    fillWithNulls(inputRDD, inputType, targetSchema,
+      field2Columns.map(i => i.columnName -> i.fieldName).toMap)
+  }
+
+  private def load(rdd: RDD[Seq[Any]], columnMetadata: Seq[ColumnMetadata], writeConf: VectorEndpointConf): Unit = {
+    val insertRDD = new InsertRDD(rdd, writeConf)
+    val writer = new DataStreamWriter[Seq[Any]](writeConf, columnMetadata)
+    insertRDD.sparkContext.runJob(insertRDD, writer.write _)
+  }
+
+  /** Given an `rdd` with data types specified by `rddSchema`, try to load it to the Vector table `table`
```

Comment:
...I'm not sure if this indenting for the doc is ok (see `wiki`)

you should reformat it; apply everywhere it fits please